expression/builtin_arithmetic.go

```diff
@@ -334,6 +340,107 @@ func (s *builtinArithmeticMinusIntSig) evalInt(row []types.Datum) (val int64, is
 	return a - b, false, nil
 }
 
+type arithmeticDivideFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *arithmeticDivideFunctionClass) getFunction(args []Expression, ctx context.Context) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	tpA, tpB := args[0].GetType(), args[1].GetType()
+	tcA, tcB := numericContextResultType(tpA), numericContextResultType(tpB)
+	if tcA == types.ClassReal || tcB == types.ClassReal {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpReal, tpReal, tpReal)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4RealOrDecimal(bf.tp, tpA, tpB, true)
+		sig := &builtinArithmeticDivideRealSig{baseRealBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	} else if tcA == types.ClassDecimal || tcB == types.ClassDecimal {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDecimal, tpDecimal, tpDecimal)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4RealOrDecimal(bf.tp, tpA, tpB, false)
+		sig := &builtinArithmeticDivideDecimalSig{baseDecimalBuiltinFunc{bf}}
+		return sig.setSelf(sig), nil
+	} else {
+		bf, err := newBaseBuiltinFuncWithTp(args, ctx, tpDecimal, tpInt, tpInt)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+		setFlenDecimal4Int(bf.tp, tpA, tpB)
```

Comment:
```
drop table if exists t;
create table t(a bigint, b bigint);
insert into t select 1, 3;
select a/b from t;
```

result is:

```sql
mysql> select a/b from t;select a/b from t;
Field   1:  `a/b`
Catalog:    `def`
Database:   ``
Table:      ``
Org_table:  ``
Type:       NEWDECIMAL
Collation:  binary (63)
Length:     25
Max_length: 6
Decimals:   4
Flags:      BINARY NUM


+--------+
| a/b    |
+--------+
| 0.3333 |
+--------+
1 row in set (0.00 sec)
```

seems we can not use `setFlenDecimal4Int` to set `Flen` and `Decimal` for result type of function `/`